### PR TITLE
CPPBitmapLayer class added

### PR DESCRIPTION
**Pull request by triedal**
- Added C++ wrapper class for BitmapLayer. 
- Reorganized file structure to separate examples from source code.
- Updated README

### CODE REVIEW

File: examples/basics_cpp/pebble.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include "pebble.h"
```
Comment (fahhem) [inline examples/basics_cpp/pebble.cpp]: Why is this file copied? Can you update the wscript to include this from the src/ directory instead?

**Comment by fahhem** [comment]
Thanks so much! I'd love to accept this, but I'd prefer not to have the pebble, malloc, etc files copied instead of included and linked against

**Comment by triedal** [comment]
Sure! I noticed you mentioned on your blog that you could include the files by symlinking to the wscript. I'm not too sure how that works exactly. Could you give me some pointers as to what I need to do?

**Comment by fahhem** [comment]
If you're on Windows, you can't symlink, but on Mac and Linux you can. Just do "ln -s ../../src/wscript ." in the example directories and once for each file (just replace wscript).

Maybe I should come up with something else, but I'm pretty sure the main wscript file has changed anyway now with the new patforms and all.

**Comment by triedal** [comment]
I'm on a Mac. I'll give it a shot :)

**Comment by triedal** [comment]
Are there any further changes I need to make?

**Comment by fahhem** [comment]
Nope!

**Pull request merged**